internal/store/interfaces.go

```diff
@@ -59,6 +67,19 @@ type Store interface {
 	Sync(ctx context.Context) error
 }
 
+// NewStore is used for persisting and querying gateways and routes
+type NewStore interface {
```

Comment:
This is the new store interface which `store/store.go` already fulfills. It will fully replace the `Store` interface above in a followup PR